### PR TITLE
[FSTORE-556] Add examples about include_primary_key and include_event_time

### PR DESCRIPTION
**Pull request by kennethmhc**
This PR adds/fixes/changes...
- Add examples about include_primary_key and include_event_time
- Add doc string for extra_filter

JIRA Issue: -

Priority for Review: -

Related PRs: -

**How Has This Been Tested?**

- [ ] Unit Tests
- [ ] Integration Tests
- [ ] Manual Tests on VM

**Checklist For The Assigned Reviewer:**

```
- [ ] Checked if merge conflicts with master exist
- [ ] Checked if stylechecks for Java and Python pass
- [ ] Checked if all docstrings were added and/or updated appropriately
- [ ] Ran spellcheck on docstring
- [ ] Checked if guides & concepts need to be updated
- [ ] Checked if naming conventions for parameters and variables were followed
- [ ] Checked if private methods are properly declared and used
- [ ] Checked if hard-to-understand areas of code are commented
- [ ] Checked if tests are effective
- [ ] Built and deployed changes on dev VM and tested manually
- [x] (Checked if all type annotations were added and/or updated appropriately)
```

### CODE REVIEW

File: python/hsfs/feature_group.py [hunk 1]
```diff
@@ -116,8 +116,35 @@ def select_all(self, include_primary_key=True, include_event_time=True):
...
+                            Feature("ts", type="bigint"),
+                            Feature("f1", type="date"),
+                            Feature("f2", type="double")
+                            ],
+                    primary_key=["id"],
+                    event_time="ts")
+
+            query = fg.select_all()
+            query.features
+            # [Feature('id', ...), Feature('ts', ...), Feature('f1', ...), Feature('f2', ...)]
```
Comment (moritzmeister) [inline python/hsfs/feature_group.py:136]: This won't work as you have to call `fg.insert` to actually create the FG. I think we can keep it short as in the example above and just assume the feature group exists and do `fs.get_or_create_feature_group(...)`

File: python/hsfs/feature_group.py [hunk 2]
```diff
@@ -116,8 +116,35 @@ def select_all(self, include_primary_key=True, include_event_time=True):
...
+            query = fg.select_all(include_primary_key=False, include_event_time=False)
+            query.features
+            # [Feature('f1', ...), Feature('f2', ...)]
             ```
 
+        # Arguments
+            include_primary_key: boolean, optional. If True, include primary key of the feature group
+            to the feature list. Defaults to True.
+            include_event_time: boolean, optional. If True, include event time of the feature group
+            to the feature list. Defaults to True.
```
Comment (moritzmeister) [inline python/hsfs/feature_group.py:147]: In general you shouldn't need to describe the types manually if you put the type annotations in the method signature. For example, see here:
image


Should be indented.
```suggestion
            include_primary_key: boolean, optional. If True, include primary key of the feature group
                to the feature list. Defaults to True.
            include_event_time: boolean, optional. If True, include event time of the feature group
                to the feature list. Defaults to True.
```

File: python/hsfs/feature_group.py [hunk 3]
```diff
@@ -138,19 +165,31 @@ def select_all(self, include_primary_key=True, include_event_time=True):
...
             # connect to the Feature Store
             fs = ...
 
             # get the Feature Group instance
-            fg = fs.get_or_create_feature_group(...)
-
-            # construct the query
-            fg.select(['date', 'weekly_sales', 'is_holiday'])
+            from hsfs.feature import Feature
+            fg = fs.create_feature_group(
```
Comment (moritzmeister) [inline python/hsfs/feature_group.py:178]: Same comment as above.

File: python/hsfs/feature_view.py [hunk 4]
```diff
@@ -711,6 +711,8 @@ def create_training_data(
             description: A string describing the contents of the training dataset to
                 improve discoverability for Data Scientists, defaults to empty string
                 `""`.
+            extra_filter: List of `Filter` or `FilterLogic`. Optional. Attach additional filters
```
Comment (moritzmeister) [inline python/hsfs/feature_view.py:714]: Same as above regarding types.

**Review by moritzmeister — CHANGES_REQUESTED** [review]

File: python/hsfs/feature_group.py (hunk 1 above)
Comment (kennethmhc) [reply python/hsfs/feature_group.py:136]: This example works without actually creating a fg. 
I think it is good to show the feature definition, including name, primary key and event time for demonstrating how `include_primary_key` and `include_event_time` work.

Comment (moritzmeister) [reply python/hsfs/feature_group.py:140]: aha ok but you cannot execute the query

**Review by moritzmeister — APPROVED** [review]

**Pull request merged**